treeherder/seta/update_job_priority.py

```diff
@@ -0,0 +1,353 @@
+            return None
+        else:
+            LOG.info("We're going to fetch new runnable jobs data.")
+            # We never store the output of runnable api but the minimal data we need
+            data = self.sanitized_data(self.query_runnable_jobs(repo_name=repo_name, task_id=task_id))
+
+            if data:
+                # Store the sanitized data to disk for inspection
+                with open(path, 'w') as f:
+                    json.dump(data, f, indent=2, sort_keys=True)
+
+            return data
+
+    def update_job_priority_table(self):
+        """Use it to update the job priority table with data from the runnable api."""
+        # XXX: We are assuming that the jobs accross 'mozilla-inbound', 'autoland' and 'fx-team'
+        #      are equivalent. This could cause issues in the future
+        data = self.query_sanitized_data(repo_name='mozilla-inbound')
+
+        if data:
+            self._update_table(data)
+        else:
+            LOG.warning('We received an empty data set')
+            return
+
+    def query_runnable_jobs(self, task_id, repo_name='mozilla-inbound'):
+        url = RUNNABLE_API.format(repo_name, task_id)
+        try:
+            data = retry(requests.get, args=(url, ), kwargs={'headers': HEADERS}).json()
+            if data:
+                # A lot of code components still rely on the file being on disk
+                with open(get_runnable_jobs_path(), 'w') as f:
+                    json.dump(data, f, indent=2, sort_keys=True)
+
+            return data
+        except Exception as e:
+            LOG.warning("We failed to get runnablejobs via %s" % url)
+            LOG.warning(str(e))
+            return None
+
+    def parse_testtype(self, build_system_type, job_type_name, platform_option, refdata):
+        # TODO: figure out how to ignore build, lint, etc. jobs
+        if build_system_type == 'buildbot':
+            return refdata.split(' ')[-1]
+        else:
+            # e.g. "desktop-test-linux64/opt-reftest-8",
+            # taskcluster test types
+            testtype = job_type_name
+            if testtype.startswith('[funsize'):
+                return None
+
+            testtype = testtype.split('/opt-')[-1]
+            testtype = testtype.split('/debug-')[-1]
+
+            # this is plain-reftests for android
+            testtype = testtype.replace('plain-', '')
+
+            # SETA/Ouija do not care about builds, or other jobs like lint, etc.
+            testtype = testtype.replace(' Opt', 'build')
+            testtype = testtype.replace(' Debug', 'build')
+            testtype = testtype.replace(' Dbg', 'build')
+            testtype = testtype.replace(' (opt)', 'build')
+            testtype = testtype.replace(' PGO Opt', 'build')
+            testtype = testtype.replace(' Valgrind Opt', 'build')
+            testtype = testtype.replace(' Artifact Opt', 'build')
+            testtype = testtype.replace(' (debug)', 'build')
```

Comment:
If mozci (or part of it) was part of the code, I could make this code not needing updating.
mozci has a query_metadata() function that allow us to take a ref_data_name and determine all metadata.

I could also add such metadata to the runnable jobs API.

Let me know what you prefer me to do:
A) just update the list
B) bring mozci support

mozci support could also be filed and left for later to prevent scope creep.